### ISSUE TITLE
Linux PV / HVM documentation

### ISSUE THREAD

**Issue by iuriguilherme** [QubesOS/qubes-issues#1332]
Taking from #1175,

One of the first things I wanted to do on Qubes was to create VMs using Kali, Whonix and other similar distros. Actually, trying to have Kali and Tails on the same environment led me to the Whonix site, which in turn introduced the name "Qubes" for me to begin with.

The [page on setting up a Windows 7 HVM](https://www.qubes-os.org/en/doc/windows-appvms/) was the first step I had to take when trying to figure out the proper path through the docs, until I finally found out the [HVM Create](https://www.qubes-os.org/en/doc/hvm-create/) page, which I originally expected was one of the first things on the [main doc page](https://www.qubes-os.org/en/doc/).

Shouldn't similar newbie specific guides like this be available for Linux users as well? I don't think it is proper to assume that a Linux user has much more knowledge on the matter that a Windows user. I have background of administering Xen servers and I am learning from scratch how to deal with domU on Qubes, seems totally different to me.

My personal opinion is that stuff like Whonix should be in a Paravirtualized domain. However, I would like to use ElementaryOS, Kali and similar specific distros like a user would want to use Windows 7, for instance. So for the general purpose, one of the main things on the main doc page, in my opinion, should be something like "Installing and Using Linux-based PV/HVM AppVMs" or similar.

**Edit:** just went through the whole [HVM Create](https://www.qubes-os.org/en/doc/hvm-create/) page and figured out it basically teaches only Windows 7. There should be at least a "Tip: this also works for linux based isos!", or such. Mentioning Ubuntu in the meanwhile seem not enough, I had to go through many pages and now have to reverse read all doing `s/Windows 7/Any OS/g` inside my mind.

**Comment by axon-qubes** [comment]
Thank you for submitting this issue. However, your description is rather convoluted. I'll try to summarize, and you can tell me if I've understood you correctly or not.

1. Move the HVM Create page higher up on the main doc page.
2. Rewrite the HVM Create page so that it also applies to Linux.
3. Write a page explaining how to make PV domains for other Linux distros.

**Comment by desci** [comment]
The HVM Create page is labeled "Creating and Using HVM and Windows Domains (Qubes R2+)" on the main doc page (or ToC), that's why i missed it in the first place, I didn't understand its purpose, so I clicked on other links that seemed more like what I was trying to achieve.

So both your 1 and 2 really leads to:

1. Make it obvious for a lazy eye that Qubes can be used to install Linux distros.

And the number 3 would boil down to:

1. How? Here's How.

Which would lead to Paravirtualization with some kind of link in the beginning leading to HVM.

I should have brought this to the mailing list first :stuck_out_tongue: if you help me, I may edit the original post.

**Comment by axon-qubes** [comment]
> 1. Make it obvious for a lazy eye that Qubes can be used to install Linux distros.
> 2. How? Here's How.

It sounds like we can accomplish both tasks by creating a new page with the title "Installing other Linux distros" which contains the content you described. Agreed?

> if you help me, I may edit the original post.

I'm not sure what you mean. Help you how?

**Comment by adrelanos** [comment]
Few comments.

* Related: document TemplateBasedHVM domains #1341
* https://www.qubes-os.org/en/doc/hvm-create/ : Since https://github.com/QubesOS/qubes-issues/issues/712 was implemented, no more command line use should be required to insert the installation CD? Is this worth a separate ticket or fine here?
* Any other uses of command line that is no longer necessary?
* https://www.qubes-os.org/en/doc/hvm-create/ could use a table of contents.

**Comment by axon-qubes** [comment]
> * https://www.qubes-os.org/en/doc/hvm-create/ : Since #712 was implemented, no more command line use should be required to insert the installation CD? Is this worth a separate ticket or fine here?
> * Any other uses of command line that is no longer necessary?

For better or worse, many of our docs use CLI commands instead of QVMM. In some cases, they were written before QVMM had the relevant functionality. In other cases, it's just easier to provide the CLI commands than to provide step-by-step screenshots of QVMM. It's also more precise to say `qvm-create win7 --hvm --label green` than just to say, "Create a green Windows 7 HVM." Of course, one could simply say both, or provide the CLI command and also say, "You can also do this in Qubes Manager."

**Comment by mfc** [comment]
I think the CLI thread should be a separate ticket, as it is not specific to the PV / HVM documentation.

**Comment by axon-qubes** [comment]
@adrelanos:
> https://www.qubes-os.org/en/doc/hvm-create/ could use a table of contents.

[Done.](https://github.com/QubesOS/qubes-doc/commit/480b348f4ac9a98a3a205531900f77d9ec6c62ed)

**Comment by adrelanos** [comment]
Did you consider using an automated TOC generator?

**Comment by axon-qubes** [comment]
Yes, that would be great! Can you point me in the right direction?

**Comment by adrelanos** [comment]
Axon:
> Yes, that would be great! Can you point me in the right direction?

I am not experienced with jekyll.

From:
http://jekyllrb.com/docs/plugins/

> Table of Content Generator: Generate the HTML code containing a table
of content (TOC), the TOC can be customized in many way, for example you
can decide which pages can be without TOC.

https://github.com/dafi/jekyll-toc-generator

**Comment by axon-qubes** [comment]
Thanks, @adrelanos!

However:
> Github pages can't use plugins
> If you host your jekyll pages on github you can't run plugins, in this scenario you can use a full javascript solution using TOC Generator for Markdown

Looking into the latter now.

**Comment by axon-qubes** [comment]
Hm, can't seem to get [this](https://github.com/dafi/tocmd-generator) to work despite trying everything suggested in the README, on other guides online, and every other variation I could think of. Maybe our site handles scripts differently or something.

**Comment by desci** [comment]
@axon-qubes

> It sounds like we can accomplish both tasks by creating a new page with the title "Installing other Linux distros" which contains the content you described. Agreed?

Indeed. And that's what I could help with, i think.

> I'm not sure what you mean. Help you how?

I have no idea why I wrote that, in that way.

**Comment by kulinacs** [comment]
I would like to work on this, but I would like to get some input before I rework everything.
Would it make more sense to have the ["Creating and using HVM (fully virtualized) domains"](https://www.qubes-os.org/doc/hvm/) rewritten to read more OS agnostic, to better convey that the installation process works with any ISO, not just Windows and Ubuntu. It could then link to the already Windows Specific page ["Installing and using Windows-based AppVMs"](https://www.qubes-os.org/doc/windows-appvms/) which could be rebranded as "Windows in Qubes OS", with a Windows specific (pictures and all) example of installing it as an HVM, with a link back to the generalized documentation for clarity. This would leave room in the OS Agnostic Version of ["Creating and using HVM (fully virtualized) domains"](https://www.qubes-os.org/doc/hvm/) for multiple examples at the end, possibly Tails, Open/FreeBSD, etc.

**Comment by andrewdavidwong** [comment]
> Would it make more sense to have the "Creating and using HVM (fully virtualized) domains" rewritten to read more OS agnostic, to better convey that the installation process works with any ISO, not just Windows and Ubuntu.

Yes, but I think it's already pretty close to that particular goal. There's still a lot of room for improvement for that page, however.

> It could then link to the already Windows Specific page "Installing and using Windows-based AppVMs" which could be rebranded as "Windows in Qubes OS", with a Windows specific (pictures and all) example of installing it as an HVM, with a link back to the generalized documentation for clarity.

Do you mean the ability to install Windows as both an HVM and an HVM Template? Yes, I think it's important to clarify the difference there. But it also perhaps points to the need for more general documentation on HVM Templates.

> This would leave room in the OS Agnostic Version of "Creating and using HVM (fully virtualized) domains" for multiple examples at the end, possibly Tails, Open/FreeBSD, etc. and allow the merging of Linux HVM tips that seems small and out of place by itself.

Sounds good.

**Comment by kulinacs** [comment]
To better clarify, this is what I'm planning:

[Creating and using HVM (fully virtualized) domains](https://www.qubes-os.org/doc/hvm/)

- Document HVM Templates, which addresses #1341
- Explicitly mention in the first section that it works with any ISO to avoid the confusion seen in the original issue
- Move "Installing Windows Tools" section into related links, seems out of place

[Installing and using Windows-based AppVMs](https://www.qubes-os.org/doc/windows-appvms/)

- Clarify the difference between HVMs and HVM templates

[Linux HVM Tips](https://www.qubes-os.org/doc/linux-hvm-tips/)

(I feel the documentation for general HVMs is already quite long, and adding to it may make it hard to use)

- Add sections for popular distros like Ubuntu, Arch
- Expand on Qubes HVM tools, pulling the mailing list information into the documentation

**Comment by andrewdavidwong** [comment]
Sounds good to me.

**Comment by Thovthe** [comment]
**Update on where this is at as far as I can see:**

> [Creating and using HVM (fully virtualized) domains](https://www.qubes-os.org/doc/hvm/)

- Links out to [Windows Tools](https://www.qubes-os.org/doc/windows-tools/)
  - Should link directly to the template section and needs rephrasing in doc/HVM
- This was taken care
  -  Section needs further rephrasing
- This got moved out to the same Windows Tools as above
  - Section should be moved below Assigning PCI

> [Installing and using Windows-based AppVMs](https://www.qubes-os.org/doc/windows-tools/#using-template-based-windows-appvms-qubes-r2-beta-)
 
- Really good information that gets us half way there
  - I'm left wondering what the exact architecture differences are and if any HVM can be turned into an HVM-template

> [Linux HVM Tips](https://www.qubes-os.org/doc/linux-hvm-tips/)

- I'm not sure exactly what kulinacs was intending to do with this page but it doesn't appear to have happened.
<br></br>

**Thoughts:**

The VM documentation is already unwieldy, but it's not really to do with the volume of information.  These pages + the one on [building templates](https://www.qubes-os.org/doc/building-non-fedora-template) need some rework, reorg and maybe a rethinking if we want to solve this documentation flow issue.  I have also had trouble with following the steps to get VMs with other distros set up how I want as I've had to piece it together from 5+ qubes docs and other wikis.  Maybe we outline the ends a user might want and then flow them through the documentation based on the goal.
<br></br>

**Anyone else have thoughts about how to move forward with improving this?**

**Comment by unman** [comment]
I believe that most of these issues have been dealt with in one way or another.
The linux-hvm-tips have been moved to external documentation in the Forum.
